Dockerfile

```diff
@@ -1,19 +1,31 @@
 FROM python:3.8
 
-WORKDIR /
+ARG UID
+ARG GID
+
+# set environment variables
+ENV PYTHONDONTWRITEBYTECODE 1
+ENV PYTHONUNBUFFERED 1
 
+WORKDIR /
 # install requirements
 COPY ./requirements.txt requirements.txt
 RUN pip install -r requirements.txt
 
 # create a new user and use it.
-RUN useradd -M -u 1001 nonrootuser
-USER nonrootuser
+RUN mkdir -p /home/iceesuser
+RUN groupadd --system -g $GID iceesuser && useradd --system -g iceesuser --shell /bin/bash -u $UID --home /home/iceesuser iceesuser
 
+WORKDIR /home/iceesuser
 # set up API things
 COPY ./icees_api icees_api
 COPY ./main.sh main.sh
 COPY ./examples examples
 
+RUN chown -R iceesuser:iceesuser /home/iceesuser
+
+USER iceesuser
```

Comment:
If we move `USER iceesuser` up to before we copy over all files, do we have to run the `chown` command?

Reply:
Yes, we had to. That was actually what I did initially, but since all files were still owned by root, I added this ```chown``` which fixed the permission issue.